tests/subsys/net/lib/lwm2m/lwm2m_rd_client/testcase.yaml

```diff
@@ -0,0 +1,6 @@
+tests:
+  subsys.net.lib.lwm2m_rd_client:
+    tags: lwm2m_engine
+    platform_allow: native_posix
```

Comment:
`depends_on: netif` to limit test execution only to platforms that support networking